Add flag, environment var to report times in UTC

jrick opened the pull request:
If a logger is created with the slog.LUTC flag, or the LOGFLAGS
environment variable specifies the option UTC, report all timestamps
in the log output in UTC rather than using the local time zone.

chappjc commented on log.go:
```diff
@@ -265,6 +271,9 @@ func callsite(flag uint32) (string, int) {
 	t := time.Now() // get as early as possible
+	if b.flag&LUTC != 0 {
+		t = t.UTC()
+	}
```
Will need to update `printf` below too I believe.

jrick replied:
good catch

jrick committed:
Add flag, environment var to report times in UTC

If a logger is created with the slog.LUTC flag, or the LOGFLAGS
environment variable specifies the option UTC, report all timestamps
in the log output in UTC rather than using the local time zone.

chappjc reviewed (approved):
Concept ACK..  You use `WithFlags` to create a backend options function that you provide to `NewBackend`?  Alt to using the env var?

jrick commented:
That's right, here's a play snippet showing how WithFlag can be used https://play.golang.org/p/dQyqtZ1ice3

chappjc reviewed (approved):
This flags approach seems to work well in dcrdex https://github.com/decred/dcrdex/pull/657/files#diff-d6d9aeccfe8ca6b182938340dbb8c79b
I don't anticipate a need to set the location to anything other than UTC or local, so https://github.com/decred/slog/pull/4 is probably not ideal.
